### PR TITLE
#19302  once password is lost we regenerate the keyStore

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
if for any reason password gets lost we need to regenerate the keystore.
Plus some cleanup + AppsCache Test included in the Maisuite

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/security/apps/SecretsStoreKeyStoreImplTest.java [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.util.Config;
 import com.dotmarketing.util.UUIDGenerator;
+import java.security.KeyStore;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/security/apps/SecretsStoreKeyStoreImplTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'java.security.KeyStore'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6221863)

File: dotCMS/src/integration-test/java/com/dotcms/security/apps/SecretsStoreKeyStoreImplTest.java [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+import java.security.KeyStoreException;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/security/apps/SecretsStoreKeyStoreImplTest.java:14]: Codacy found an issue: [Avoid unused imports such as 'java.security.KeyStoreException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6221863)

File: dotCMS/src/main/java/com/dotcms/security/apps/SecretsStoreKeyStoreImpl.java [hunk 3]
```diff
@@ -2,24 +2,35 @@
...
 import com.liferay.util.FileUtil;
 import com.rainerhahnekamp.sneakythrow.Sneaky;
+import io.vavr.control.Try;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/SecretsStoreKeyStoreImpl.java:33]: Codacy found an issue: [Avoid unused imports such as 'java.security.GeneralSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6221863)

File: dotCMS/src/main/java/com/dotcms/security/apps/SecretsStoreKeyStoreImpl.java [hunk 4]
```diff
@@ -2,24 +2,35 @@
...
 import com.dotmarketing.business.CacheLocator;
 import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotRuntimeException;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.UtilMethods;
 import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.language.LanguageException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/security/apps/SecretsStoreKeyStoreImpl.java:22]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.portal.language.LanguageException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6221863)

**Review by erickgonzalez — APPROVED** [review]

**Comment by erickgonzalez** [comment]
@fabrizzio-dotCMS Just one question, shouldn't this PR be against `release-5.3.9`??

**Comment by fabrizzio-dotCMS** [comment]
> @fabrizzio-dotCMS Just one question, shouldn't this PR be against `release-5.3.9`??



> @fabrizzio-dotCMS Just one question, shouldn't this PR be against `release-5.3.9`??

That's a great question I did it against master cuz that's what we agreed yesterday with @wezell but maybe you're right and this should be against release-5.3.9

**Review by jdotcms — APPROVED** [review]

**Comment by fabrizzio-dotCMS** [comment]
closing this PR in favor of this https://github.com/dotCMS/core/pull/19328

**Pull request closed without merging**